jetty-client/src/main/java/org/eclipse/jetty/client/util/InputStreamResponseListener.java

```diff
@@ -346,13 +347,13 @@ private IOException toIOException(Throwable failure)
         public void close() throws IOException
         {
             List<Callback> callbacks;
-            synchronized (lock)
+            try (AutoLock.WithCondition l = lock.lock())
             {
                 if (closed)
                     return;
                 closed = true;
                 callbacks = drain();
-                lock.notifyAll();
+                l.signalAll();
```

Comment:
I'm almost certain `signalAll()` is wrong, but that's outside the scope of this PR.